staging/src/k8s.io/sample-controller/README.md

```diff
@@ -23,6 +23,14 @@ Changes should not be made to these files manually, and when creating your own
 controller based off of this implementation you should not copy these files and
 instead run the `update-codegen` script to generate your own.
 
+## Details
+
+The sample controller uses [client-go library](https://github.com/kubernetes/client-go/tree/master/tools/cache) extensively.
+The interaction points of the controller with various mechanisms from this library are depicted in following diagram.
+
+![Alt text](./client-go-controller-interaction.jpeg.?raw=true "Kubernetes client-go and Custom Controller")
```

Comment:
This doesn't appear to be rendering correctly for me 🤔 do relative links work for images?

Reply:
Perhaps we can also move the image into a subdirectory too? Either `images/` or `docs/images/`?

Reply:
Doesn't render correctly for me as well. `docs/images/` sgtm.

@devdattakulkarni Your write-up was really great as well! Would you be willing to add it here? :)

Reply:
Thanks for the comments.

@munnerz: I will move the image to docs/images

@nikhita: Thanks for the feedback. I will include the write-up as well.